programs/editor/plugins/bella/Bella.js

```diff
@@ -0,0 +1,230 @@
+/**
+ * Copyright (C) 2014-2015 KO GmbH <copyright@kogmbh.com>
+ *
+ * @licstart
+ * This file is part of WebODF.
+ *
+ * WebODF is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License (GNU AGPL)
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * WebODF is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with WebODF.  If not, see <http://www.gnu.org/licenses/>.
+ * @licend
+ *
+ * @source: http://www.webodf.org/
+ * @source: https://github.com/kogmbh/WebODF/
+ */
+
+/*global define, window, ops*/
+
+define("webodf/editor/plugins/bella/Bella", [
+        "webodf/editor/plugins/bella/Actions",
+        "webodf/editor/plugins/bella/CatHerder",
+        "webodf/editor/plugins/bella/DocumentValidator",
+        "webodf/editor/plugins/bella/Random",
+        "webodf/editor/plugins/bella/SimpleStatusReporter",
+        "webodf/editor/plugins/bella/WrappedSessionController"
+    ],
+    function(Actions, CatHerder, DocumentValidator, Random, SimpleStatusReporter, WrappedSessionController) {
+    "use strict";
+    var instanceCount = 0,
+        webodf = window; // WebODF doesn't support AMD yet...
+
+    /**
+     * @constructor
+     * @implements {core.Destroyable}
+     * @param {!Object=} config Configuration options to pass to fuzzer
+     *      automatically generated
+     */
+    function Bella(config) {
+        var self = this,
+            MAX_UNDO_STATES = 1000,
+            seed,
+            random,
+            currentSessionController,
+            actionSources = [],
+            scheduledTask,
+            documentValidator,
+            undoStateCount = 0,
+            reporter,
+            errorHandler,
+            state = {
+                name: config.name,
+                startedAt: undefined,
+                actions: 0,
+                executedOperations: 0,
+                seed: undefined
+            };
+
+        function countOperations() {
+            undoStateCount += 1; // TODO count states NOT operations
+            state.executedOperations += 1;
+        }
+
+        /**
+         * Workaround a TrivialUndoManager bug with cursor positions not being correctly restored
+         */
+        function saveCurrentCursorPositions() {
+            var odtDocument = currentSessionController.getSession().getOdtDocument(),
+                members = odtDocument.getMemberIds(),
+                ops = [];
+
+            members.forEach(function (memberId) {
```

Comment:
Please explain more why OpMoveCursor operations are created. What is the bug that is being worked around?